### PR TITLE
add CentOS stream 9 support

### PR DESCRIPTION
**Pull request by guits**
This adds the resquired changes in order to support CentOS stream 9.

Also, this bumps the Ansible version support to 2.15

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
 # These are Python requirements needed to run ceph-ansible main
-ansible-core>=2.12,<2.13
+ansible-core>=2.15,<2.16,!=2.9.10
```
Comment (asm0deuz) [inline requirements.txt:2]: why keep !=2.9.10? with the previous conditions there is no chance to have that version installed

Comment (guits) [reply requirements.txt:2]: it doesn't hurt, but yes, there's no need to keep it

File: tests/requirements.txt [hunk 2]
```diff
@@ -2,7 +2,7 @@
 testinfra
 pytest-xdist
 pytest
-ansible-core>=2.12,<2.13,!=2.9.10
+ansible-core>=2.15,<2.16,!=2.9.10
```
[the remark this answers is not in the document]
Comment (guits) [reply tests/requirements.txt:5]: same answer as above :sweat_smile:

**Comment by guits** [comment]
jenkins test centos-non_container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-lvm_osds

**Comment by guits** [comment]
jenkins test centos-container-lvm_batch

**Comment by guits** [comment]
jenkins test centos-container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-all_in_one

**Comment by guits** [comment]
jenkins test centos-container-collocation

**Comment by guits** [comment]
jenkins test centos-container-collocation

**Comment by guits** [comment]
jenkins test centos-container-purge

**Comment by guits** [comment]
jenkins test centos-non_container-all_daemons

**Comment by clwluvw** [comment]
jenkins test centos-container-lvm_batch

**Comment by guits** [comment]
jenkins test centos-container-lvm_batch

**Comment by guits** [comment]
jenkins test centos-container-lvm_batch

**Comment by guits** [comment]
jenkins test centos-container-lvm_batch

**Comment by guits** [comment]
jenkins test centos-container-lvm_batch

**Comment by guits** [comment]
jenkins test centos-non_container-lvm_batch

**Comment by guits** [comment]
jenkins test centos-non_container-lvm_osds

File: library/ceph_crush_rule.py [hunk 3]
```diff
@@ -230,27 +232,24 @@ def main():
...
             rc, cmd, out, err = exec_command(module, remove_rule(module, container_image=container_image))  # noqa: E501
             changed = True
         else:
             rc = 0
             out = "Crush Rule {} doesn't exist".format(name)
-
-    elif state == "info":
-        rc, cmd, out, err = exec_command(module, get_rule(module, container_image=container_image))  # noqa: E501
+    else:
+        pass
```
Comment (clwluvw) [inline library/ceph_crush_rule.py:252]: I guess these two lines are unnecessary

File: roles/ceph-mon/tasks/deploy_monitors.yml [hunk 4]
```diff
@@ -88,42 +88,92 @@
...
+    add_key: "{{ admin_secret if admin_secret != 'admin_secret' else omit }}"
+  delegate_to: "{{ groups[mon_group_name][0] }}"
+  run_once: true
   environment:
     CEPH_CONTAINER_IMAGE: "{{ ceph_docker_registry + '/' + ceph_docker_image + ':' + ceph_docker_image_tag if containerized_deployment | bool else None }}"
     CEPH_CONTAINER_BINARY: "{{ container_binary }}"
-  register: create_custom_admin_secret
-  no_log: "{{ no_log_on_ceph_key_tasks }}"
+  no_log: False
+#  no_log: "{{ no_log_on_ceph_key_tasks }}"
```
Comment (clwluvw) [inline roles/ceph-mon/tasks/deploy_monitors.yml:112]:
```suggestion
  no_log: "{{ no_log_on_ceph_key_tasks }}"
```

File: library/ceph_crush_rule.py (hunk 3 above)
Comment (guits) [reply library/ceph_crush_rule.py:252]: indeed, they are not needed

File: roles/ceph-mon/templates/ceph-mon.service.j2 [hunk 5]
```diff
@@ -54,7 +54,7 @@ ExecStart=/usr/bin/{{ container_binary }} run --rm --name ceph-mon-%i \
   --entrypoint=/usr/bin/ceph-mon \
   {{ceph_docker_registry }}/{{ ceph_docker_image }}:{{ ceph_docker_image_tag }} \
   -f --default-log-to-file=false --default-log-to-stderr=true \
-  -i {{ monitor_name }} --mon-data /var/lib/ceph/mon/{{ cluster }}-{{ monitor_name }} --public-addr {{ _current_monitor_address }}
+  -i {{ monitor_name }} --mon-data /var/lib/ceph/mon/{{ cluster }}-{{ monitor_name }} --public-addr {{ _current_monitor_address }} --mon-initial-members mon0
```
Comment (clwluvw) [inline roles/ceph-mon/templates/ceph-mon.service.j2:57]: `mon0` looks wrong I guess here.

File: roles/ceph-validate/tasks/check_system.yml [hunk 6]
```diff
@@ -6,8 +6,8 @@
 
 - name: fail on unsupported ansible version
   fail:
-    msg: "Ansible version must be 2.12!"
-  when: ansible_version.minor|int != 12
+    msg: "Ansible version must be 2.14!"
+  when: ansible_version.minor|int != 14
```
Comment (clwluvw) [inline roles/ceph-validate/tasks/check_system.yml:10]: Just a note to go with 2.16 for this PR

File: site-container.yml.sample [hunk 7]
```diff
@@ -174,6 +174,60 @@
...
+      ceph_config:
+        action: set
+        who: "global"
+        option: "{{ item.key }}"
+        value: "{{ item.value }}"
+      environment:
+        CEPH_CONTAINER_IMAGE: "{{ ceph_docker_registry + '/' + ceph_docker_image + ':' + ceph_docker_image_tag if containerized_deployment | bool else None }}"
+        CEPH_CONTAINER_BINARY: "{{ container_binary }}"
+      when: inventory_hostname == ansible_play_hosts_all | last
+      with_dict: "{{ ceph_conf_overrides['global'] }}"
```
Comment (clwluvw) [inline site-container.yml.sample:217]: We need to take `ceph_conf_overrides` into consideration for all daemons as well as some modules need to have configs before startup.

**Comment by guits** [comment]
jenkins test centos-container-lvm_batch

**Comment by guits** [comment]
jenkins test centos-non_container-switch_to_containers

**Comment by clwluvw** [comment]
jenkins test centos-container-update

**Comment by guits** [comment]
jenkins test centos-container-update

**Comment by guits** [comment]
jenkins test centos-container-subset_update

**Comment by guits** [comment]
jenkins test centos-non_container-lvm_osds

**Comment by guits** [comment]
linter will be addressed separately

**Pull request merged**